File: Groupe2/Android/Appsurde/app/src/main/java/com/neopixl/appsurde/network/UrlBuilder.java
package com.neopixl.appsurde.network;

/**
 * Created by Yvan Moté on 20/06/2017.
 */

public class UrlBuilder {

    // http://api.tvmaze.com/shows/73/episodes

    private enum Environment {
        DEV, PROD, TEST
    }

    // final = constante, non modifiable
    private static final String BASE_URL =
            "http://api.tvmaze.com";

    private static final String DEV_BASE_URL =
            "http://dev.api.tvmaze.com";

    private static Environment CURRENT_ENVIRONMENT
            = Environment.PROD;

    public static String getEpisodesUrl(int showId) {
        return getBaseUrl()+"/shows/"+showId+"/episodes";
    }

    private static String getBaseUrl() {
        switch (CURRENT_ENVIRONMENT) {
            case PROD: return BASE_URL;
            case DEV: return DEV_BASE_URL;
            default: return BASE_URL;
        }
    }


}

File: Groupe2/iOS/AppHero/AppHero/UrlBuilder.swift
//
//  UrlBuilder.swift
//  AppHero
//
//  Created by savepanda on 23/06/2017.
//  Copyright © 2017 Neopixl. All rights reserved.
//

import Foundation

class UrlBuilder {
    
    // http://api.tvmaze.com/shows/73/episodes
    
    private static let baseUrl = "https://api.tvmaze.com/"
    
    static func allEpisodesUrl(showId: Int) -> String {
        return baseUrl+"shows/\(showId)/episodes"
    }
    
}
